enterprise/api/v1/views.py

```diff
@@ -999,6 +1000,37 @@ class EnterpriseCustomerBrandingConfigurationViewSet(EnterpriseReadOnlyModelView
     ordering_fields = FIELDS
     lookup_field = 'enterprise_customer__slug'
 
+    @action(methods=['patch'], detail=True, permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_access_admin_dashboard', fn=lambda request, enterprise_uuid: enterprise_uuid)
+    def update_branding(self, request, enterprise_uuid):
+        """
+        POST /enterprise/api/v1/enterprise-customer-branding/update_branding
+        """
+        try:
+            enterprise_customer = models.EnterpriseCustomer.objects.get(uuid=enterprise_uuid)
+            if len(models.EnterpriseCustomerBrandingConfiguration.objects.all()):
+                branding_config = models.EnterpriseCustomerBrandingConfiguration.objects.get(
+                    enterprise_customer=enterprise_customer)
+            else:
+                branding_config = models.EnterpriseCustomerBrandingConfiguration(
+                    enterprise_customer=enterprise_customer)
+
+            if 'logo' in request.data:
+                branding_config.logo = request.data['logo']
+            if 'primary_color' in request.data:
+                branding_config.primary_color = request.data['primary_color']
+            if 'secondary_color' in request.data:
+                branding_config.secondary_color = request.data['secondary_color']
+            if 'tertiary_color' in request.data:
+                 branding_config.tertiary_color = request.data['tertiary_color']
+            branding_config.save()
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception(
+                'Error with updating branding configuration'
+            )
+            return Response("Error with updating branding configuration", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
this should probably be some flavor of http 500, since this is a broad server side exception handler. if the user had made some incorrect input, we'd expect a 400 with an error as to how they could correct their request.